### ISSUE TITLE
[win32] ERROR_WIDGET_DISPOSED in StyledText when using Accessibility

### ISSUE THREAD

**Issue by SyntevoAlex** [eclipse-platform/eclipse.platform.swt#448]
**Describe the bug**
When user runs an Accessibility based program together with SWT `StyledText`, and `StyledText` gets disposed, SWT may then fail with exceptions like
```
org.eclipse.swt.SWTException: code=24: Widget is disposed
	at org.eclipse.swt.SWT.error(SourceFile:4918)
	at org.eclipse.swt.SWT.error(SourceFile:4833)
	at org.eclipse.swt.SWT.error(SourceFile:4804)
	at org.eclipse.swt.widgets.Widget.error(SourceFile:450)
	at org.eclipse.swt.widgets.Widget.checkWidget(SourceFile:369)
	at org.eclipse.swt.custom.StyledText.getCharCount(SourceFile:4022)
	at org.eclipse.swt.custom.StyledText$7.getCharacterCount(SourceFile:6824)
	at org.eclipse.swt.accessibility.Accessible.getCharacterCount(SourceFile:5016)
	at org.eclipse.swt.accessibility.Accessible.get_nCharacters(SourceFile:4649)
	at org.eclipse.swt.accessibility.Accessible$5.method17(SourceFile:375)
	at org.eclipse.swt.internal.ole.win32.COMObject.callback17(SourceFile:219)
	at org.eclipse.swt.internal.win32.OS.PeekMessage
	at org.eclipse.swt.widgets.Display.readAndDispatch(SourceFile:3646)
```

The problem is affected by the following factors:
* It seems (but I haven't really investigated) that only `IAccessible2` is affected.
* SWT supports `org.eclipse.swt.accessibility.UseIA2=false` system property, which also avoids the bug in the scenarios I tested.
* Since Windows 10 Version 1903, Windows's UI Automation now automatically connects to `IAccessible2`. That is, UI Automation based programs now trigger the bug.
* Since Win11, Windows UI Automation seems to automatically test HWND to be valid and if is not, it returns `0x80040201` (`EVENT_E_ALL_SUBSCRIBERS_FAILED`) without invoking SWT, thus avoiding the bug.

**To Reproduce**
One way to reproduce is to use Grammarly application on Win10. Open some shell with `StyledText` and close it quickly, roughly within 300ms or so (timing may vary depending on hardware, I guess).

**Expected behavior**
No exceptions.

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [ ] All OS
- - [x] Windows
- - [ ] Linux
- - [ ] macOS 

2. Additional OS info (e.g. OS version, Linux Desktop, etc)
As explained above, the bug is most visible on Win10 since Version 1903, but not on Win11.
However, if a 3rd party program will use `IAccessible2` directly, bug will probably occur on any Windows version.

**Workaround (or) Additional context**
`org.eclipse.swt.accessibility.UseIA2=false` system property

**Comment by SyntevoAlex** [comment]
I'm already working on a patch.

**Comment by SyntevoAlex** [comment]
Patch is ready for review.
